### ISSUE TITLE
High CPU usage when JDBC driver not found

### ISSUE THREAD

**Issue by ruseel** [testcontainers/testcontainers-java#1378]
Almost similar with #230, but In this case symptom is high CPU usage(smashing InspectContainerCmdExec). 

```
2019-04-10 16:30:12.006 DEBUG   --- [     ducttape-0] c.g.d.core.exec.InspectContainerCmdExec  : GET: OkHttpWebTarget(okHttpClient=org.testcontainers.shaded.okhttp3.OkHttpClient@53d102a2, baseUrl=http://docker.socket/, path=[/containers/f949e1eaf71732d083b7ac8751210643c379ec434f77fde806518aeda2eb8a35/json], queryParams={})
2019-04-10 16:30:12.010 TRACE   --- [     ducttape-0] o.r.ducttape.unreliables.Unreliables     : Retrying lambda call on attempt 1159
2019-04-10 16:30:12.010 DEBUG   --- [     ducttape-0] c.g.d.core.command.AbstrDockerCmd        : Cmd: f949e1eaf71732d083b7ac8751210643c379ec434f77fde806518aeda2eb8a35,false,com.github.dockerjava.core.exec.InspectContainerCmdExec@2cd79a32
2019-04-10 16:30:12.010 DEBUG   --- [     ducttape-0] c.g.d.core.exec.InspectContainerCmdExec  : GET: OkHttpWebTarget(okHttpClient=org.testcontainers.shaded.okhttp3.OkHttpClient@53d102a2, baseUrl=http://docker.socket/, path=[/containers/f949e1eaf71732d083b7ac8751210643c379ec434f77fde806518aeda2eb8a35/json], queryParams={})
2019-04-10 16:30:12.014 TRACE   --- [     ducttape-0] o.r.ducttape.unreliables.Unreliables     : Retrying lambda call on attempt 1160
2019-04-10 16:30:12.014 DEBUG   --- [     ducttape-0] c.g.d.core.command.AbstrDockerCmd        : Cmd: f949e1eaf71732d083b7ac8751210643c379ec434f77fde806518aeda2eb8a35,false,com.github.dockerjava.core.exec.InspectContainerCmdExec@598e1db8
2019-04-10 16:30:12.015 DEBUG   --- [     ducttape-0] c.g.d.core.exec.InspectContainerCmdExec  : GET: OkHttpWebTarget(okHttpClient=org.testcontainers.shaded.okhttp3.OkHttpClient@53d102a2, baseUrl=http://docker.socket/, path=[/containers/f949e1eaf71732d083b7ac8751210643c379ec434f77fde806518aeda2eb8a35/json], queryParams={})
2019-04-10 16:30:12.019 TRACE   --- [     ducttape-0] o.r.ducttape.unreliables.Unreliables     : Retrying lambda call on attempt 1161
2019-04-10 16:30:12.019 DEBUG   --- [     ducttape-0] c.g.d.core.command.AbstrDockerCmd        : Cmd: f949e1eaf71732d083b7ac8751210643c379ec434f77fde806518aeda2eb8a35,false,com.github.dockerjava.core.exec.InspectContainerCmdExec@53f76db3
2019-04-10 16:30:12.019 DEBUG   --- [     ducttape-0] c.g.d.core.exec.InspectContainerCmdExec  : GET: OkHttpWebTarget(okHttpClient=org.testcontainers.shaded.okhttp3.OkHttpClient@53d102a2, baseUrl=http://docker.socket/, path=[/containers/f949e1eaf71732d083b7ac8751210643c379ec434f77fde806518aeda2eb8a35/json], queryParams={})
```

Steps to reproduce 

> - make project WITHOUT 'com.mysql.jdbc.Driver' 
> - use `MySQLContainer`
> - run 
> 

Symptom

> CPU usage is up to 30~40% 
> o.r.ducttape.unreliables.Unreliables     : Retrying lambda call on attempt 1161` is printed in`while(true)`
> 

---

https://github.com/testcontainers/testcontainers-java/blob/2da17b9855f855b3ae6ea5b9f07f8cb6080dc2fa/modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java#L125-L146

If `createConnection()` raises Exception, `isRunning()` is called in succession by `retryUntilSuccess`.

**Comment by jannis-baratheon** [comment]
The cause seems to be that [Unreliables#retryUntilSuccess](https://github.com/rnorth/duct-tape/blob/2a1c5be9f2ef3f16bf036cec8752a170d130b61e/src/main/java/org/rnorth/ducttape/unreliables/Unreliables.java#L31) only expect an `Exception` to be thrown by the poll-lambda whereas a missing driver probably throws an `Error`. This gets propagated to the executor where most probably it's swallowed leading to an infinite busy-loop somewhere.

**Comment by rnorth** [comment]
This should be solved by #1434, so I shall close.
